tonic/models/vic/grid_params.py

```diff
@@ -55,88 +86,108 @@ def __init__(self, nlayers=3, snow_bands=5):
-                                       ('fs_active', np.array([12 * nlayers
-                                                               + 16])),
-                                       ])
-
-        self.snow_param = OrderedDict([('cellnum', np.array([0])),
-                                       ('AreaFract', np.arange(1,
-                                                               snow_bands
-                                                               + 1)),
-                                       ('elevation', np.arange(snow_bands + 1,
-                                                               2 * snow_bands
-                                                               + 1)),
-                                       ('Pfactor', np.arange(2 * snow_bands
-                                                             + 1,
-                                                             3 * snow_bands
-                                                             + 1))])
-
+                                       ('c', np.array([8]))])
+
+        varnames = ['expt', 'Ksat', 'phi_s', 'init_moist']
+        i = 9
+        for var in varnames:
+            self.soil_param[var] = np.arange(i, i + nlayers)
+            i += nlayers
+
+        self.soil_param['elev'] = np.array([i])
+        i += 1
+        self.soil_param['depth'] = np.arange(i, i + nlayers)
+        i += nlayers
+        self.soil_param['avg_T'] = np.array([i])
+        i += 1
+        self.soil_param['dp'] = np.array([i])
+        i += 1
+
+        varnames = ['bubble', 'quartz', 'bulk_density', 'soil_density']
+        if organic_fract:
+            varnames.append(['organic', 'bulk_dens_org', 'soil_dens_org'])
+        for var in varnames:
+            self.soil_param[var] = np.arange(i, i + nlayers)
+            i += nlayers
+
+        self.soil_param['off_gmt'] = np.array([i])
+        i += 1
+
+        varnames = ['Wcr_FRACT', 'Wpwp_FRACT']
+        for var in varnames:
+            self.soil_param[var] = np.arange(i, i + nlayers)
+            i += nlayers
+
+        varnames = ['rough', 'snow_rough', 'annual_prec']
+        for var in varnames:
+            self.soil_param[var] = np.array([i])
+            i += 1
+
+        self.soil_param['resid_moist'] = np.arange(i, i + nlayers)
+        i += nlayers
+
+        self.soil_param['fs_active'] = np.array([i])
+        i += 1
+
+        if spatial_frost:
+            self.soil_param['frost_slope'] = np.array([i])
+            i += 1
+
+        if spatial_snow:
+            self.soil_param['max_snow_distrib_slope'] = np.array([i])
+            i += 1
+
+        if july_tavg_supplied:
+            self.soil_param['July_Tavg'] = np.array([i])
+            i += 1
+
+
+        # Snow Parameters
+        self.snow_param = OrderedDict([('cellnum', np.array([0]))])
+        varnames = ['AreaFract', 'elevation', 'Pfactor']
+        i = 1
+        for var in varnames:
```

Comment:
Nothing wrong here but `for var in ['AreaFract', 'elevation', 'Pfactor']` is cleaner.